spatialos-sdk/src/worker/schema.rs

```diff
@@ -103,22 +109,23 @@ pub trait Field {
         }
     }
 
-    fn get_list(object: &SchemaObject, field: FieldId) -> Vec<Self::RustType> {
+    fn get_list(object: &SchemaObject, field: FieldId) -> Result<Vec<Self::RustType>> {
         let count = Self::count(object, field);
-        let mut result = Vec::with_capacity(count);
-        for index in 0..count {
-            result.push(Self::index(object, field, index));
-        }
-        result
+        (0..count)
+            .map(|index| Self::index(object, field, index))
+            .collect()
     }
 
     fn has_update(update: &SchemaComponentUpdate, field: FieldId) -> bool;
 
-    fn get_update(update: &SchemaComponentUpdate, field: FieldId) -> Option<Self::RustType> {
+    fn get_update(
+        update: &SchemaComponentUpdate,
+        field: FieldId,
+    ) -> Result<Option<Self::RustType>> {
         if Self::has_update(update, field) {
-            Some(Self::get_or_default(update.fields(), field))
+            Self::get(update.fields(), field)).map(Some)
```

Comment:
```suggestion
            Self::get(update.fields(), field).map(Some)
```

Sorry I had the syntax slightly wrong in my original suggestion :grimacing:

Reply:
Ah yeah, just fixed that :smile: